www/main/viewer/ShabadDeck/ShabadDeck.jsx

```diff
@@ -209,16 +209,22 @@ function ShabadDeck() {
 
   useEffect(() => {
     if (activeVerseId && akhandpatt) {
-      const verseDOM = verseRefs.current[activeVerseId];
+      // Add a delay to ensure all verses are rendered and refs are populated
+      const scrollTimeout = setTimeout(() => {
+        const verseDOM = verseRefs.current[activeVerseId];
 
-      if (verseDOM) {
-        verseDOM.scrollIntoView({
-          behavior: 'smooth',
-          block: 'center',
-        });
-      }
+        if (verseDOM) {
+          verseDOM.scrollIntoView({
+            behavior: 'smooth',
+            block: 'center',
+          });
+        }
+      }, 500); // Increased delay to allow all verses to render
```

Comment:
@amitojsingh Can we implement this without setTimeout. The setTimeout is uncertain, can we instead have a useEffect or something similar that runs when all verses are rendered.